### PR TITLE
Remove old projects tree provider

### PR DESCRIPTION
**Pull request by kisantia**
- Removed old projects tree provider because now those are kept track of in project workspace extension
- fixed tests failing and causing build errors from these changes and tree refresh fix #12692
- Commented out some data sources code for now (isn't being used anyway) because a circular dependency between dataSources.ts and sqlConnectionStringSource.ts started causing the extension to not activate

### CODE REVIEW

File: extensions/sql-database-projects/src/controllers/projectController.ts [hunk 1]
```diff
@@ -35,21 +35,19 @@ import { ISystemDatabaseReferenceSettings, IDacpacReferenceSettings, IProjectRef
...
-	constructor(projTreeViewProvider: SqlDatabaseProjectTreeViewProvider) {
-		this.projectTreeViewProvider = projTreeViewProvider;
+	constructor() {
 		this.netCoreTool = new NetCoreTool();
 		this.buildHelper = new BuildHelper();
 	}
 
-	public refreshProjectsTree() {
-		this.projectTreeViewProvider.load(this.projects);
+	public refreshProjectsTree(treeProvider: vscode.TreeDataProvider<any>) {
```
Comment (alanrenmsft) [inline extensions/sql-database-projects/src/controllers/projectController.ts:49]: I'd prefer to accept a workspaceTreeItem as the parameter, shortly I'll be working on partial tree refreshing, knowing the tree item would be helpful. also, you don't have to do context.treeDataProvider when calling this method :)

File: extensions/sql-database-projects/src/models/dataSources/dataSources.ts [hunk 2]
```diff
@@ -53,11 +52,11 @@ export async function load(dataSourcesFilePath: string): Promise<DataSource[]> {
 	// TODO: do we have a construct for parsing version numbers?
 	switch (rawJsonContents.version) {
 		case '0.0.0':
-			const dataSources: DataSourceFileJson = rawJsonContents as DataSourceFileJson;
```
Comment (alanrenmsft) [inline extensions/sql-database-projects/src/models/dataSources/dataSources.ts:56]: can this be removed?

File: extensions/sql-database-projects/src/test/projectController.test.ts [hunk 3]
```diff
@@ -198,13 +198,13 @@ describe('ProjectsController', function (): void {
...
 
-				const projController = new ProjectsController(new SqlDatabaseProjectTreeViewProvider());
+				const projController = new ProjectsController();
 
-				await projController.delete(projTreeRoot.children.find(x => x.friendlyName === 'UpperFolder')!.children[0] /* LowerFolder */);
-				await projController.delete(projTreeRoot.children.find(x => x.friendlyName === 'anotherScript.sql')!);
-				await projController.delete(projTreeRoot.children.find(x => x.friendlyName === 'Script.PreDeployment1.sql')!);
-				await projController.delete(projTreeRoot.children.find(x => x.friendlyName === 'Script.PreDeployment2.sql')!);
-				await projController.delete(projTreeRoot.children.find(x => x.friendlyName === 'Script.PostDeployment1.sql')!);
+				await projController.delete(createWorkspaceTreeItem(projTreeRoot.children.find(x => x.friendlyName === 'UpperFolder')!.children[0]) /* LowerFolder */);
```
Comment (alanrenmsft) [inline extensions/sql-database-projects/src/test/projectController.test.ts:203]: last night I was playing with this and I did exactly the same thing, even the method name is the same. LOL

**Review by alanrenmsft — APPROVED** [review]

File: extensions/sql-database-projects/src/models/dataSources/dataSources.ts (hunk 2 above)
Comment (kisantia) [reply extensions/sql-database-projects/src/models/dataSources/dataSources.ts:56]: @Benjin is it ok to remove this data sources code or should I just leave it commented out? Is the plan for data sources to be in the sql database projects extension or in the data workspaces extension in the future?

**Pull request merged**